File: dotnet/ThirtyFour.UI.Tests/Window/UrlMatcherTests.cs
﻿using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using OpenQA.Selenium;
using ThirtyFour.UI.Window;

namespace ThirtyFour.UI.Tests.Window
{
    [TestClass]
    public class UrlMatcherTests : BaseTestSuite
    {

        [TestMethod]
        public void ExceptionThrownWhenNoMatchingUrl()
        {
            string currentHandle = driver.CurrentWindowHandle;

            try
            {
                new UrlMatcher("shjksjd").MatchWindow(driver, 4);
                Assert.Fail("An exception should have been thrown");
            }
            catch (Exception e)
            {
                Assert.AreEqual("Unable to find Window", e.Message);
            }
            finally
            {
                QuitExtraDrivers(currentHandle);
            }
        }

        [TestMethod]
        public void WindowSwitchedWhenMatchingUrl()
        {
            string currentHandle = driver.CurrentWindowHandle;

            try
            {

                driver.FindElement(By.LinkText("Popup")).Click();
                new UrlMatcher(@"https://github.com/danhumphrey/ThirtyFour.UI").MatchWindow(driver, 4);
                Assert.AreEqual("https://github.com/danhumphrey/ThirtyFour.UI", driver.Url);
            }
            finally
            {
                QuitExtraDrivers(currentHandle);
            }
        }


        [TestMethod]
        public void WindowSwitchedWhenMatchingUrlOfDelayedWindow()
        {
            string currentHandle = driver.CurrentWindowHandle;

            try
            {
                driver.FindElement(By.LinkText("Delayed Popup")).Click();
                new UrlMatcher(@"https://github.com/danhumphrey/ThirtyFour.UI").MatchWindow(driver, 5);
                Assert.AreEqual("https://github.com/danhumphrey/ThirtyFour.UI", driver.Url);
            }
            finally
            {
                QuitExtraDrivers(currentHandle);
            }
        }
    }
}

File: java/src/test/java/com/github/danhumphrey/thirtyfour/ui/test/window/UrlMatcherTests.java
package com.github.danhumphrey.thirtyfour.ui.test.window;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

import com.github.danhumphrey.thirtyfour.ui.test.BaseTestSuite;
import com.github.danhumphrey.thirtyfour.ui.window.TitleMatcher;
import com.github.danhumphrey.thirtyfour.ui.window.UrlMatcher;

public class UrlMatcherTests extends BaseTestSuite {

	@Test
	public void testExceptionThrownWhenNoMatchingUrl() {

		WebDriver driver = getDriver();
		String currentHandle = driver.getWindowHandle();

		try {
			new UrlMatcher("shjksjd").matchWindow(driver, 1000);
			Assertions.fail("An exception should have been thrown");
		} catch (Exception e) {
			Assertions.assertEquals("Unable to find a matching window", e.getMessage());
		} finally {
			quitExtraDrivers(currentHandle);
		}
	}

	@Test
    public void WindowSwitchedWhenMatchingUrl() throws Exception
    {
    	WebDriver driver = getDriver();
		String currentHandle = driver.getWindowHandle();

		driver.findElement(By.linkText("Popup")).click();
		
		try {
	        new UrlMatcher("https://github.com/danhumphrey/ThirtyFour.UI").matchWindow(driver, 1000);
	        Assertions.assertEquals("https://github.com/danhumphrey/ThirtyFour.UI", driver.getCurrentUrl());
		} finally {
			quitExtraDrivers(currentHandle);
		}
        
    }

	@Test
    public void testWindowSwitchedWhenMatchingUrlOfDelayedWindow() throws Exception
    {
    	WebDriver driver = getDriver();
		String currentHandle = driver.getWindowHandle();
		
        driver.findElement(By.linkText("Delayed Popup")).click();
        
        try {
	        new UrlMatcher("https://github.com/danhumphrey/ThirtyFour.UI").matchWindow(driver, 5000);
	        Assertions.assertEquals("https://github.com/danhumphrey/ThirtyFour.UI", driver.getCurrentUrl());
		} finally {
			quitExtraDrivers(currentHandle);
		}
    }
}
